### ISSUE TITLE
mongodb://localhost:27017 not connecting

### ISSUE THREAD

**Issue by drsimplegraffiti** [Automattic/mongoose#12366]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.5.3

### Node.js version

v18.0.0

### MongoDB server version

5.0.8

### Description

mongodb://localhost:27017 not connecting on compass

### Steps to Reproduce

/* Connecting to the database. */
const mongoose = require('mongoose');

const connectDB = async () => {
  await mongoose.connect("mongodb://localhost:27017", () => {
    console.log('connected to db');
  });
};
 
module.exports = connectDB;

### Connected to db on success

> connected to db

**Comment by Uzlopak** [comment]
node 18 uses  ipv6 under the hood. So localhost is not pointing to 127.0.0.1 but to :::1. Use the IP instead of localhost.

**Comment by hasezoey** [comment]
from what i can tell it is since node 17, see https://github.com/nodejs/node/issues/40537

**Comment by akash4-sys** [comment]
This issue is still not fixed.

**Comment by hasezoey** [comment]
> This issue is still not fixed.

if you are using nodejs over 17, then it is not something mongoose can fix (without manually handling ip resolving), so mongoose documentation was changed to not reference `localhost` anymore but `127.0.0.1` instead.

TL;DR: for anyone searching:
NodeJS 17 has changed that `localhost` points to a IPv6 address instead of a IPv4 address, the workaround is to replace `localhost` with `127.0.0.1` or make the mongodb instance listen on the IPv6 address via the [`--ipv6` commandline option](https://www.mongodb.com/docs/manual/core/security-mongodb-configuration/) (which is not on by default currently) or `--bind_ip_all`

**Comment by sophiacanja** [comment]
I fixed this problem by installing nvm (node version manager) which allowed me to use an older version of node (v14.15.5). So I did not have to replace localhost with 127.0.0.1
This is the site i referenced https://blog.logrocket.com/how-switch-node-js-versions-nvm/#:~:text=Switching%20among%20Node.js%20Version,into%20either%20version%20we%20need.

**Comment by vkarpov15** [comment]
That works, but you will likely have to switch to using 127.0.0.1 sooner or later, because Node 14 is already EOL
